sw/device/silicon_creator/lib/manifest.h

```diff
@@ -301,13 +307,12 @@ inline rom_error_t manifest_check(const manifest_t *manifest) {
 inline manifest_digest_region_t manifest_digest_region_get(
     const manifest_t *manifest) {
   enum {
-    // TODO(#17824): This should depend on `CREATOR_SW_CFG_SIGVERIFY_SPX_EN`.
     kDigestRegionOffset = offsetof(manifest_t, usage_constraints) +
                           sizeof(manifest->usage_constraints),
   };
   return (manifest_digest_region_t){
       .start = (const char *)manifest + kDigestRegionOffset,
-      .length = manifest->length - kDigestRegionOffset,
+      .length = manifest->signed_region_end - kDigestRegionOffset,
```

Comment:
Would it make sense to add some kind of sanity check here that `signed_region_end >= length`?

Reply:
Great point, I'll added a new error code `kErrorManifestBadSignedRegion` and added this check to `manifest_check()`.